ground/gcs/src/plugins/pathplanner/flightdatamodel.cpp

```diff
@@ -443,46 +487,80 @@ void FlightDataModel::readFromFile(QString fileName)
+    }
+    QDomNode node = hlist.at(0).firstChild();
     while (!node.isNull()) {
         QDomElement e = node.toElement();
         if (e.tagName() == "waypoint") {
             QDomNode fieldNode=e.firstChild();
-            data=new PathPlanData;
+            data = new PathPlanData;
+            double wpLLA[3];
+            double wpNED[3];
             while (!fieldNode.isNull()) {
                 QDomElement field = fieldNode.toElement();
                 if (field.tagName() == "field") {
-                    if(field.attribute("name")=="altitude")
-                        data->altitude=field.attribute("value").toDouble();
-                    else if(field.attribute("name")=="description")
+                    if(field.attribute("name") == "down")
+                        wpNED[2] = field.attribute("value").toDouble();
+                    else if(field.attribute("name") == "description")
                         data->wpDescription=field.attribute("value");
-                    else if(field.attribute("name")=="latitude")
-                        data->latPosition=field.attribute("value").toDouble();
-                    else if(field.attribute("name")=="longitude")
-                        data->lngPosition=field.attribute("value").toDouble();
-                    else if(field.attribute("name")=="altitude")
-                        data->altitude=field.attribute("value").toDouble();
-                    else if(field.attribute("name")=="velocity")
-                        data->velocity=field.attribute("value").toFloat();
-                    else if(field.attribute("name")=="mode")
+                    else if(field.attribute("name") == "north")
+                        wpNED[0] = field.attribute("value").toDouble();
+                    else if(field.attribute("name") == "east")
+                        wpNED[1] = field.attribute("value").toDouble();
+                    else if(field.attribute("name") == "velocity")
+                        data->velocity = field.attribute("value").toFloat();
+                    else if(field.attribute("name") == "mode")
                         data->mode=field.attribute("value").toInt();
-                    else if(field.attribute("name")=="mode_params")
-                        data->mode_params=field.attribute("value").toFloat();
-                    else if(field.attribute("name")=="is_locked")
-                        data->locked=field.attribute("value").toInt();
+                    else if(field.attribute("name") == "mode_params")
+                        data->mode_params = field.attribute("value").toFloat();
+                    else if(field.attribute("name") == "is_locked")
+                        data->locked = field.attribute("value").toInt();
                 }
                 fieldNode=fieldNode.nextSibling();
             }
+            Utils::CoordinateConversions().NED2LLA_HomeLLA(HomeLLA, wpNED, wpLLA);
```

Comment:
if you want to defer this to the code cleanup it's fine, but probably it would be best to set the values initially to known values (e.g. NAN)  to verify all fields were set. Otherwise error out.